codegen/sections_test.go

```diff
@@ -0,0 +1,68 @@
+package codegen
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestHeader(t *testing.T) {
+	const (
+		pack  = "testpackage"
+		title = "test title"
+	)
+	var (
+		imprt   = []*ImportSpec{&ImportSpec{Path: "test"}}
+		imports = append(imprt, &ImportSpec{Path: "other"})
+	)
+	cases := map[string]struct {
+		Title    string
+		Imports  []*ImportSpec
+		Expected string
+	}{
+		"no-title":      {Expected: noTitleHeader},
+		"title":         {Title: title, Expected: titleHeader},
+		"single-import": {Imports: imprt, Expected: singleImportHeader},
+		"many-imports":  {Imports: imports, Expected: manyImportsHeader},
+	}
+	for k, tc := range cases {
+		buf := new(bytes.Buffer)
+		s := Header(tc.Title, "testpackage", tc.Imports)
+		s.Render(buf)
+		actual := buf.String()
+		if actual != tc.Expected {
+			t.Errorf("%s: got %#v, expected %#v", k, actual, tc.Expected)
+		}
+	}
+}
+
+const (
```

Comment:
These `const` should be in scope of `TestHeader()`, right? We have to take care to the duplication of that when we add some tests in this package.